src/schoology-extractor/schoology_extractor/schoology_extract_facade.py

```diff
@@ -235,7 +235,24 @@ def get_attendance_events(
 
         events = self._client.get_attendance(section_id)
 
-        events_df = attendanceMap.map_to_udm(events, section_associations)
+        def _sync_wrapper(data: pd.DataFrame):
+            """ Attendance_events entity has a more complex mapping, for most
+            entities, sync process running before mapping is fine, but considering
+            the process involved for this entity, the mapper for attendance events
+            will accept an additional callable for sync process that will run when
+            the mapping process is ready
+            """

```

Comment:
I don't know if it is a good idea to have it as a docstring or I should just clarify it as a comment in the PR and remove it from the code. Please let me know your thoughts @stephenfuqua

Reply:
Docstrings are used for Intellisense or for auto-generated documentation. If you think the comment should show up in those two situations, then I would use the docstring - but then also document the argument and return type. Otherwise I would just do a regular comment with `#` at the beginning of every line.

The signature needs the return type.

Reply:
But this is probably fine.